lib/sisimai/bite/email/mxlogic.rb

```diff
@@ -0,0 +1,253 @@
+        }x,
+        hostunknown: %r{(?>
+             all[ ](?:
+                 host[ ]address[ ]lookups[ ]failed[ ]permanently
+                |relevant[ ]MX[ ]records[ ]point[ ]to[ ]non[-]existent[ ]hosts
+                )
+            |Unrouteable[ ]address
+            )
+        }x,
+        mailboxfull: %r{(?:
+             mailbox[ ]is[ ]full:?
+            |error:[ ]quota[ ]exceed
+            )
+        }x,
+        notaccept: %r{(?:
+             an[ ]MX[ ]or[ ]SRV[ ]record[ ]indicated[ ]no[ ]SMTP[ ]service
+            |no[ ]host[ ]found[ ]for[ ]existing[ ]SMTP[ ]connection
+            )
+        }x,
+        systemerror: %r{(?>
+             delivery[ ]to[ ](?:file|pipe)[ ]forbidden
+            |local[ ]delivery[ ]failed
+            |LMTP[ ]error[ ]after[ ]
+            )
+        }x,
+        contenterror: %r{
+            Too[ ]many[ ]["]Received["][ ]headers
+        }x,
+      }
+      ReDelayed = %r{(?:
+         retry[ ]timeout[ ]exceeded
+        |No[ ]action[ ]is[ ]required[ ]on[ ]your[ ]part
+        |retry[ ]time[ ]not[ ]reached[ ]for[ ]any[ ]host[ ]after[ ]a[ ]long[ ]failure[ ]period
+        |all[ ]hosts[ ]have[ ]been[ ]failing[ ]for[ ]a[ ]long[ ]time[ ]and[ ]were[ ]last[ ]tried
+        |Delay[ ]reason:[ ]
+        |Message[ ].+[ ](?:has[ ]been[ ]frozen|was[ ]frozen[ ]on[ ]arrival[ ]by[ ])
+        )
+      }x
+      Indicators = Sisimai::Bite::Email.INDICATORS
+
+      def description; return 'McAfee SaaS'; end
+      def smtpagent;   return Sisimai::Bite.smtpagent(self); end
+      # X-MX-Bounce: mta/src/queue/bounce
+      # X-MXL-NoteHash: ffffffffffffffff-0000000000000000000000000000000000000000
+      # X-MXL-Hash: 4c9d4d411993da17-bbd4212b6c887f6c23bab7db4bd87ef5edc00758
+      def headerlist;  return ['X-MXL-NoteHash', 'X-MXL-Hash', 'X-MX-Bounce']; end
+      def pattern;     return Re0; end
+
+      # Parse bounce messages from MXLogic
+      # @param         [Hash] mhead       Message headers of a bounce email
+      # @options mhead [String] from      From header
+      # @options mhead [String] date      Date header
+      # @options mhead [String] subject   Subject header
+      # @options mhead [Array]  received  Received headers
+      # @options mhead [String] others    Other required headers
+      # @param         [String] mbody     Message body of a bounce email
+      # @return        [Hash, Nil]        Bounce data list and message/rfc822
+      #                                   part or nil if it failed to parse or
+      #                                   the arguments are missing
+      def scan(mhead, mbody)
+        return nil unless mhead
+        return nil unless mbody
+
+        match  = 0
+        match += 1 if mhead['x-mx-bounce']
+        match += 1 if mhead['x-mxl-hash']
+        match += 1 if mhead['x-mxl-notehash']
+        match += 1 if mhead['subject'] =~ Re0[:subject]
+        match += 1 if mhead['from']    =~ Re0[:from]
+        return nil if match.zero?
+
```

Comment:
__[rubocop]__
* Metrics/AbcSize: Assignment Branch Condition size for scan is too high. [113.4/30] [:ref](http://c2.com/cgi/wiki?AbcMetric)

<!-- Comment by [SideCI](https://sideci.com) -->